source/common/http/http1/codec_impl.cc

```diff
@@ -188,7 +189,30 @@ void StreamEncoderImpl::encodeData(Buffer::Instance& data, bool end_stream) {
   }
 }
 
-void StreamEncoderImpl::encodeTrailers(const HeaderMap&) { endEncode(); }
+void StreamEncoderImpl::encodeTrailers(const HeaderMap& trailers) {
+  if (chunk_encoding_) {
+    // Finalize the body
+    connection_.buffer().add(LAST_CHUNK);
+
+    // Trailers only matter if it is a chunk transfer encoding
+    // https://tools.ietf.org/html/rfc7230#section-4.4
+    trailers.iterate(
```

Comment:
See `StreamEncoderImpl::encodeHeaders`. This should be unified with the logic there which ultimately calls `encodeFormattedHeader` so that we get equivalent header casing, etc.